{{ cookiecutter.github_repo_name }}/tests/test_calculator.py

```diff
@@ -0,0 +1,17 @@
+from {{cookiecutter.package_dir_name}} import calculator
+
+
+def test_dot_product_2D():
+    """Test the dot product function in 2D."""
+    a = [1, 2]
+    b = [3, 4]
+    expected = 11
+    assert calculator.dot_product(a, b) == expected
+
+
+def test_dot_product_3D():
```

Comment:
then a better "best practice" here would be to use `@pytest.mark.paramatrize` to cover the 2D and 3D cases.

I like having two examples,, one with parametrize and one without, to show the "easier" case, then the more complicated one, but testing the 2D and 3D cases is a classic use of parametrize so I think that this example would be a good one for parametrize.

Reply:
Done in https://github.com/Billingegroup/scikit-package/pull/310